skimage/feature/tests/test_sift.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_almost_equal, assert_equal
+
+from skimage import data
+from skimage._shared.testing import test_parallel
+from skimage.feature import SIFT
+from skimage.util.dtype import _convert
+
+img = data.coins()
+
+
+@test_parallel()
+@pytest.mark.parametrize(
+    'dtype', ['float32', 'float64', 'uint8', 'uint16', 'int64']
+)
+def test_keypoints_sift(dtype):
+    _img = _convert(img, dtype)
+    detector_extractor = SIFT()
+    detector_extractor.detect_and_extract(_img)
+
+    exp_keypoint_rows = np.array([201, 41, 209, 180, 142, 182, 208, 34, 43,
+                                  64])
+    exp_keypoint_cols = np.array([264, 328, 271, 268, 326, 227, 223, 331, 345,
+                                  289])
```

Comment:
What is the source of the expected values in the new tests? Are these obtained from a third-party reference implementation or were these obtained from the implementation here and intended just to protect against unexpected future changes in behavior?

Reply:
The second. The values are obtained from the implementation.